modules/monitoring-resizing-a-persistent-storage-volume.adoc

```diff
@@ -0,0 +1,203 @@
++
+[NOTE]
+====
+You can resize the volumes for the Alertmanager, Thanos Ruler and Prometheus instances that monitor user-defined projects.
+====
++
+.. Edit the `user-workload-monitoring-config` `ConfigMap` object in the `openshift-user-workload-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-user-workload-monitoring edit configmap user-workload-monitoring-config
+----
+
+.. Update the PVC configuration for the monitoring component under `data/config.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    <component>: <1>
+      volumeClaimTemplate:
+        spec:
+          storageClassName: <storage_class> <2>
+          resources:
+            requests:
+              storage: <amount_of_storage> <3>
+----
+<1> Specify the core monitoring component.
+<2> Specify the storage class.
+<3> Specify the new size for the storage volume.
++
+The following example changes the PVC size to 100 gigabytes the Prometheus instance that monitors user-defined projects:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    prometheus:
+      volumeClaimTemplate:
+        spec:
+          storageClassName: local-storage
+          resources:
+            requests:
+              storage: 100Gi
+----
++
+The following example changes the PVC size to 20 gigabytes for Thanos Ruler:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    thanosRuler:
+      volumeClaimTemplate:
+        spec:
+          storageClassName: *local-storage*
+          resources:
+            requests:
+              storage: 20Gi
+----
++
+[NOTE]
+====
+Storage requirements for the `thanosRuler` component depend on the number of rules that are evaluated and how many samples each rule generates.
+====
+
+. Save the file to apply the changes. The pods affected by the new configuration restart automatically.
++
+[WARNING]
+====
+When you save changes to a monitoring config map, the pods and other resources in the related project might be redeployed. The running monitoring processes in that project might also be restarted.
+====
+
+. Manually patch every PVC with the updated storage request. The following example resizes the storage size for the `openshift-monitoring` Prometheus component to 100Gi:
++
+[source,terminal]
+----
+# for p in $(oc get pvc -l app.kubernetes.io/name=prometheus -o jsonpath='{range .items[*]}{.metadata.name} {end}'); do \
+  oc patch pvc/${p} --patch '{"spec": {"resources": {"requests": {"storage":"100Gi"}}}}'; \
+  done
+----
+
+. Delete the underlying StatefulSet with the `--cascade=orphan` parameter:
```

Comment:
we could delete this step, reason
1. there is not `--cascade` flag now
```
# oc version
Client Version: 4.12.0-0.nightly-2022-10-05-053337
Kustomize Version: v4.5.4
Server Version: 4.12.0-0.nightly-2022-11-10-033725
Kubernetes Version: v1.25.2+f33d98e

# oc -n openshift-monitoring get statefulset -l app.kubernetes.io/name=prometheus --cascade=orphan
error: unknown flag: --cascade
See 'oc get --help' for usage.
```
2. old pods would be removed automatically and new pods would be restarted automatically

Reply:
I think you typed the wrong command by accident, you used `get` instead of `delete`, I'm not aware of `--cascade` being deprecated

Reply:
@juzhao @JoaoBraveCoding - Can you please provide guidance on whether or not I need to change anything? Does `--cascade` need to be removed from the command? Thanks!

Reply:
we could remove the step
`Delete the underlying StatefulSet with the --cascade=orphan parameter:`
reason please see my previous comment

Reply:
I think we do really need this step. This step is specifically called out in the [prometheus-operator docs](https://prometheus-operator.dev/docs/operator/storage/#resizing-volumes) and iiuc is caused by a missing k8s feature.

And @JoaoBraveCoding  is correct, the cascade option should still be there. This is 4.11 but 4.12 should look the same:
```
jan@jflaptop ~/Downloads % oc version
Client Version: 4.11.13
Kustomize Version: v4.5.4
jan@jflaptop ~/Downloads % oc delete --help | grep cascade -C2
	If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.

    --cascade='background':
	Must be "background", "orphan", or "foreground". Selects the deletion cascading strategy for the dependents (e.g. Pods created by a ReplicationController). Defaults to background.
```

Reply:
ok, my mistake, I used `oc get` command, which should be `oc delete`, no need to remove, we need this step, lgtm
```
# oc version
Client Version: 4.12.0-0.nightly-2022-10-05-053337
Kustomize Version: v4.5.4
Server Version: 4.12.0-0.nightly-2022-11-10-033725

# oc -n openshift-monitoring delete statefulset -l app.kubernetes.io/name=prometheus --cascade=orphan
statefulset.apps "prometheus-k8s" deleted
```

Reply:
Thanks, all, for your input and thorough review comments. Very much appreciated.